lms/djangoapps/courseware/views/views.py

```diff
@@ -1485,6 +1487,34 @@ def enclosing_sequence_for_gating_checks(block):
     return None
 
 
+def _check_sequence_exam_access(request, location):
+    """
+    Checks the client request for an exam access token for a sequence.
+    Exam access is always granted at the sequence block. This method of gating is
+    only used by the edx-exams system and NOT edx-proctoring.
+    """
+    if not exams_ida_enabled(location.course_key):
+        return True
+
+    if request.user.is_staff or is_masquerading_as_specific_student(request.user, location.course_key):
```

Comment:
If is_masquerading_as_specific_student, don't we still need to check that there is an exam access token? Or is the idea that there wouldn't be a token. If so is it possible that the access for the masquerading user won't match the actual user's access?

Reply:
yeah in the case of masquerading there wouldn't be a token but we don't care because this implies they are actually a staff user. Overall there's probably a whole discussion on how to make masquerading work best with this but we haven't scoped that out yet.